### PR TITLE
Added hover state to links

### LINKED ISSUE #22: Add hover state to navigation
Based on https://github.com/wagtail/wagtail/issues/6793

The main navigation needs a hover state.  

We need to define what should the hover state do? Change link color? Show/hide underline?

### PR DESCRIPTION
**Pull request by PaarthAgarwal**
I have added teal and darker teal for hover and active state respectively. Fixes #22.

### CODE REVIEW

File: sass/_component_link.scss [hunk 1]
```diff
@@ -12,4 +12,7 @@ a {
 	&:visited {
 		color: $link-hover-color;
 	}
+    
+    &:active {
+        color: $link-active-color;
```
Comment (Stormheg) [inline sass/_component_link.scss:17]: This fails to compile, there is a closing brace missing after this line

**Comment by Stormheg** [comment]
Thanks for the quick fix!

File: sass/_component_link.scss (hunk 1 above)
Comment (PaarthAgarwal) [reply sass/_component_link.scss:17]: Sorry😶

**Comment by Stormheg** [comment]
Thank your for the PR @PaarthAgarwal! 🙏  ❤️ 

The hover state color looks fine to me and has enough contrast (4.95 according to my DevTools, so it should be accessible enough!)
image

**Remarks**
I think it would be good to also apply the hover state on `:focus`, so users who navigate using `Tab` and `Shift + Tab` can see which link they've focused on. The focus state can be triggered for easy testing using the developer tools in your browser. Here is a screenshot for Chrome DevTools 
image


Additionally, the hover state color is also applied for visited links. Which I think looks a bit odd: 
Screenshot 2022-04-04 at 7 24 18 PM

Not sure what kind of visited styling would work best, let's discuss in https://github.com/wagtail/sphinx_wagtail_theme/issues/78 🙂 

I'll leave this open for now until we decide on an appropriate color for visited links.

**Comment by PaarthAgarwal** [comment]
Thanks for the appreciation @Stormheg 🫂
We should add :focus indeed. Started discussing visited links before creating this PR😁

File: sass/_component_link.scss (hunk 1 above)
Comment (Stormheg) [reply sass/_component_link.scss:17]: No problem! Happens to all of us sometimes 😊

**Pull request closed without merging**